admittance_controller/include/admittance_controller/admittance_rule_impl.hpp

```diff
@@ -134,14 +145,14 @@ void AdmittanceRule::apply_parameters_update()
   }
 }
 
+
 bool AdmittanceRule::get_all_transforms(
   const trajectory_msgs::msg::JointTrajectoryPoint & current_joint_state,
-  const trajectory_msgs::msg::JointTrajectoryPoint & reference_joint_state)
+  const geometry_msgs::msg::Pose & reference_pose)
```

Comment:
why not add another method with the same name and different argument?

Reply:
I started that way originally - overloading both that and `AdmittanceRule::update()` function.

Because the functions are 90% the same, only one line changed, the resulting code looked messy. That is why I first opted for this "reworked" version until I get it working.

Reply:
It is good that you decided to go that way at first. Now, when we know that is working, we should look towards the polishing. If required, split `update` method to private functions that are used by both of them.